src/test/java/org/mastodon/mamut/clustering/ui/DendrogramPanelTest.java

```diff
@@ -138,6 +141,20 @@ public void testPaint()
 		assertEquals( DendrogramPanel.CLUSTER_LINE_COLOR, graphics.getColor() );
 	}
 
+	@Test
+	public void testPaintNoData()
+	{
+		DendrogramPanel< String > dendrogramPanel = new DendrogramPanel<>( null );
+		int width = 600;
+		int height = 600;
+		Image image = new BufferedImage( width, height, BufferedImage.TYPE_INT_ARGB );
+		Graphics graphics = image.getGraphics();
+		Color defaultColor = Color.WHITE;
+		graphics.setColor( defaultColor );
+		dendrogramPanel.paint( graphics );
+		assertEquals( DendrogramPanel.CLUSTER_LINE_COLOR, graphics.getColor() );
```

Comment:
This assertion test an irrelevant implementation detail. What you actually want to ensure is that the paint method does not fail. But you need an assert for sonacube. Interestingly there is a assertDoesNotThrow(...) in some unit testing libraries. But not in junit5. I would suggest to have a simple assertNotNull(graphics). And a comment the explains the purpose of the test. Like: `\\ NB: This test ensures that the paint method still works if the clustering is null`

Reply:
Thanks for letting me know `assertDoesNotThrow` exists in junit5.

I migrated all tests from junit4 to junit5: https://github.com/mastodon-sc/mastodon-deep-lineage/pull/95 and https://github.com/mastodon-sc/mastodon-deep-lineage/pull/96

I used `assertDoesNotThrow` in this test